src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala

```diff
@@ -42,17 +41,48 @@ object AFiAnalysis extends SummaryAnalysis {
 //        }
 //    }.unify
 
-
     // TODO somehow convert AFiSummary to AFiData
     import spark.implicits._
 
-    val summaryDF = AFiDF.getFeatureDataFrame(summaryRDD, spark)
+    val summaryDF = AFiDF
+      .getFeatureDataFrame(summaryRDD, spark)
+      .withColumn(
+        "gadm_id", when(col("location_id") =!= -1|| col("gadm_id").contains("null"), lit("") ).otherwise(col("gadm_id"))
+      )
+      .groupBy($"list_id", $"location_id", $"gadm_id")
+      .agg(
+        sum("natural_land_extent").alias("natural_land_extent"),
+        sum("tree_cover_loss_area").alias("tree_cover_loss_area"),
+        sum("negligible_risk_area").alias("negligible_risk_area"),
+        sum("total_area").alias("total_area"),
+        max("status_code").alias("status_code"),
+        concat_ws(", ", collect_list(when(col("location_error").isNotNull && col("location_error") =!= "", col("location_error")))).alias("location_error")
+      )
+
+
+    val gadmAgg = summaryDF
+      .filter($"location_id" === -1)
+      .groupBy($"list_id")
+      .agg(
```

Comment:
Would it be possible to move this agg to a function we can apply to both DFs? I think they're the same. That might make it easier to read.

Reply:
Good call, done.